### DISCUSSION TITLE
Recommended espresso test setup: emulator device + api combination

### DISCUSSION THREAD

**Discussion by masclot** [oppia/oppia-android#4868] [Q&A (Installation)] [answered]
Hi,

when running espresso tests, is there a recommended/mandatory setup for choosing the emulator and android version combo? Any other setup I should be aware of?

Thanks!

**Comment by BenHenning** [comment, marked as answer]
Great question @masclot! There's no current required setup, so long as it's compatible with the app. FWIW I usually use one of the Pixel XL-esque form factors with API 28-30 when running tests, though anything is fair game.

One benefit of not enforcing or even recommending a particular setup is that it helps to ensure we get more broad coverage when tests are run. However, this isn't very beginner-friendly, so this is perhaps something we could improve in our onboarding documentation (/cc @seanlip) by providing instructions for how to pick an API/form factor combination.
